samples/modules/tflite-micro/hello_world/src/main_functions.cc

```diff
@@ -104,32 +96,34 @@ void loop(void)
 	 */
 	float position = static_cast < float > (inference_count) /
 			 static_cast < float > (kInferencesPerCycle);
-	float x_val = position * kXrange;
+	float x = position * kXrange;
 
-	/* Place our calculated x value in the model's input tensor */
-	input->data.f[0] = x_val;
+	/* Quantize the input from floating-point to integer */
+	int8_t x_quantized = x / input->params.scale + input->params.zero_point;
+	/* Place the quantized input in the model's input tensor */
+	input->data.int8[0] = x_quantized;
 
 	/* Run inference, and report any error */
 	TfLiteStatus invoke_status = interpreter->Invoke();
 	if (invoke_status != kTfLiteOk) {
-		TF_LITE_REPORT_ERROR(error_reporter, "Invoke failed on x_val: %f\n",
-				     static_cast < double > (x_val));
+		TF_LITE_REPORT_ERROR(error_reporter, "Invoke failed on x: %f\n",
+				     static_cast < double > (x));
 		return;
 	}
 
-	/* Read the predicted y value from the model's output tensor */
-	float y_val = output->data.f[0];
+	/* Obtain the quantized output from model's output tensor */
+	int8_t y_quantized = output->data.int8[0];
+	/* Dequantize the output from integer to floating-point */
+	float y = (y_quantized - output->params.zero_point) * output->params.scale;
 
 	/* Output the results. A custom HandleOutput function can be implemented
 	 * for each supported hardware target.
 	 */
-	HandleOutput(error_reporter, x_val, y_val);
+  HandleOutput(error_reporter, x, y);
```

Comment:
this indentation seems incorrect.

Reply:
Good catch! Fixed in latest pushed commits.